zebra-state/src/service/finalized_state/zebra_db.rs

```diff
@@ -128,25 +120,63 @@ impl ZebraDb {
         let debug_skip_format_upgrades = read_only
             || ((cfg!(test) || cfg!(feature = "shielded-scan")) && debug_skip_format_upgrades);
 
-        // Open the database and do initial checks.
-        let mut db = ZebraDb {
-            config: Arc::new(config.clone()),
-            debug_skip_format_upgrades,
-            format_change_handle: None,
-            // After the database directory is created, a newly created database temporarily
-            // changes to the default database version. Then we set the correct version in the
-            // upgrade thread. We need to do the version change in this order, because the version
-            // file can only be changed while we hold the RocksDB database lock.
-            db: DiskDb::new(
-                config,
-                db_kind,
-                format_version_in_code,
-                network,
-                column_families_in_code,
-                read_only,
-            ),
+        let db_kind = db_kind.as_ref();
+        let column_families_in_code: Vec<_> = column_families_in_code.into_iter().collect();
+        let new_db = |format_version| {
+            ZebraDb {
+                config: Arc::new(config.clone()),
+                debug_skip_format_upgrades,
+                format_change_handle: None,
+                // After the database directory is created, a newly created database temporarily
+                // changes to the default database version. Then we set the correct version in the
+                // upgrade thread. We need to do the version change in this order, because the version
+                // file can only be changed while we hold the RocksDB database lock.
+                db: DiskDb::new(
+                    config,
+                    db_kind,
+                    &format_version,
+                    network,
+                    column_families_in_code.clone(),
+                    read_only,
+                ),
+            }
         };
 
+        match format_change.clone() {
+            DbFormatChange::Upgrade {
+                older_disk_version,
+                newer_running_version,
+            } if !debug_skip_format_upgrades
+                && newer_running_version.major == (older_disk_version.major + 1)
+                && RESTORABLE_DB_VERSIONS.contains(&newer_running_version.major) =>
+            {
+                warn!("upgrading database format to the next major version");
```

Comment:
###### Optional

I'd say this log entry is part of normal operation.

```suggestion
                info!("upgrading database format to the next major version");
```